### PR TITLE
Update versions and fix bugs

### PR DESCRIPTION
**Pull request by stevenzamborsky**
Update versions of Boundary images and the Boundary Terraform provider as well as fix deprecated issues and a bug in the docker-compose.

### CODE REVIEW

File: compose/docker-compose.yml [hunk 1]
```diff
@@ -28,7 +28,7 @@ services:
         condition: service_healthy
 
   controller:
-    image: hashicorp/boundary:0.1.1
+    image: hashicorp/boundary:0.12.0
```
Comment (stellarsquall) [inline compose/docker-compose.yml]:
```suggestion
    image: hashicorp/boundary:0.11.2
```

File: compose/docker-compose.yml [hunk 2]
```diff
@@ -17,7 +17,7 @@ services:
       retries: 5
   
   db-init:
-    image: hashicorp/boundary:0.1.1
+    image: hashicorp/boundary:0.12.0
```
Comment (stellarsquall) [inline compose/docker-compose.yml]:
```suggestion
    image: hashicorp/boundary:0.11.2
```

**Review by stellarsquall — CHANGES_REQUESTED** [review]
I think we need to install a slightly older version of the controller, so that we have a newer version to upgrade to in the tutorial?

**Review by stellarsquall — APPROVED** [review]
thank you!

**Comment by stevenzamborsky** [comment]
Agreed - good catch @stellarsquall

**Pull request merged**